### PR TITLE
Create viewmodel for creating a recipe

### PR DESCRIPTION
**Pull request by arnekuchenbecker**

### CODE REVIEW

File: app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import javax.inject.Inject
+
+@HiltViewModel
+class CreateRecipeViewModel @Inject constructor(
+    private val createRecipe: CreateRecipe
+) : ViewModel() {
+    var recipeName by mutableStateOf("")
+    var uri by mutableStateOf<Uri?>(null)
+    var description by mutableStateOf("")
+    var calculatedFor by mutableStateOf(1)
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt:45]: Why 1?

File: app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private val _ingredients = mutableStateMapOf<String, SnapshotStateList<Ingredient>>()
+    val ingredients: Map<String, List<Ingredient>>
+        get() = _ingredients
+
+    val navigateFlow = MutableStateFlow<Long?>(null)
+
+    fun createRecipe() {
+        if (recipeName.isBlank()
+            || ingredients.isEmpty()
+            || ingredients.all { (_, ingredients) -> ingredients.isEmpty() }) {
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt:57]: Shouldn't it check that there is not any empty ingredientgroup?

File: app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt (hunk 1 above)
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt:45]: We need a default value here and the other two obvious choices would be 0 or a negative number, however both of these kind of break the assumption that when calculating the amount needed to cook a recipe (later on) we can just do `<amount of ingredients> * <number of people> / <calculatedFor>`

Comment (kittyahy) [reply app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt:45]: okay

**Comment by kittyahy** [comment]
ui/viewmodel/CreateRecipeViewModel.kt:67:17 Cannot find a parameter with this name: allergen 
Doesn't build

**Review by kittyahy — APPROVED** [review]

**Pull request merged**